src/app/scripts/cac/routing/cac-routing-plans.js

```diff
@@ -0,0 +1,109 @@
+CAC.Routing.Plans = (function($, L, moment, _, UserModes, Itinerary) {
+    'use strict';
+
+    var planMode = ['WALK', 'TRANSIT'];
+    var routingUrl = '/otp/routers/default/plan';
+
+    var module = {
+        planTrip: planTrip,
+        getItineraryById: getItineraryById,
+        itineraries: {},
+        setItineraryStyles: setItineraryStyles,
+        removeItineraryLayers: removeItineraryLayers
+    };
+
+    return module;
+
+    /**
+     * Find shortest path from one point to another
+     *
+     * @param {array} coordsFrom The coords in lat-lng which we would like to travel from
+     * @param {array} coordsTo The coords in lat-lng which we would like to travel to
+     *
+     * @return {promise} The promise object which - if successful - resolves to a
+     *                   an object with itineraries
+     */
+    function planTrip(coordsFrom, coordsTo, map) {
+        var deferred = $.Deferred();
+        var urlParams = prepareParamString(coordsFrom, coordsTo);
+        var requestUrl = routingUrl + '?' + urlParams;
+        return $.ajax({
+            url: requestUrl,
+            type: 'GET',
+            contentType: 'application/json'
+        }).then(function(data) {
+            removeItineraryLayers(map);
```

Comment:
I think this is probably the wrong place for this map interaction. Do it on Pages.Map when this promise resolves, since not all planTrips() necessarily care about a map.